cmd/common/oci.go

```
@@ -0,0 +1,140 @@
+				return err
+			}
+
+			// gadget parameters that are only available after contacting the server
+			gadgetParams := make(params.Params, 0)
+
+			// Before running the gadget, we need to get the gadget info to perform
+			// different tasks like creating the parser and setting flags for the
+			// gadget's parameters.
+			actualArgs := cmd.Flags().Args()
+
+			gadgetCtx := gadgetcontext.NewSimple(context.Background(), actualArgs[0], logger.DefaultLogger())
+
+			// Fetch gadget information; TODO: this can potentially be cached
+			info, err := runtime.GetOCIGadgetInfo(gadgetCtx, nil, args)
+			if err != nil {
+				return fmt.Errorf("fetching gadget information: %w", err)
+			}
+			log.Debugf("GI %+v %v", info, err)
+
+			for _, p := range info.Params {
+				gadgetParams.Add(api.ParamToParamDesc(p).ToParam())
+			}
+
+			AddFlags(cmd, &gadgetParams, nil, runtime)
+
+			return cmd.ParseFlags(args)
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			// args from RunE still contains all flags, since we manually parsed them,
+			// so we need to manually pull the remaining args here
+			args := cmd.Flags().Args()
+
+			if len(args) == 0 {
+				if showHelp, _ := cmd.Flags().GetBool("help"); showHelp {
+					additionalMessage := "Specify the gadget image to get more information about it"
+					cmd.Long = fmt.Sprintf("%s\n\n%s", cmd.Short, additionalMessage)
+				}
+				return cmd.Help()
+			}
+
+			if showHelp, _ := cmd.Flags().GetBool("help"); showHelp {
+				return cmd.Help()
+			}
+
+			// we also manually need to check the verbose flag, as PersistentPreRunE in
+			// verbose.go will not have the correct information due to manually parsing
+			// the flags
+			checkVerboseFlag()
+
+			fe := console.NewFrontend()
+			defer fe.Close()
+
+			ctx := fe.GetContext()
+
+			gadgetCtx := gadgetcontext.NewSimple(ctx, args[0], logger.DefaultLogger())
+
+			info, err := runtime.GetOCIGadgetInfo(gadgetCtx, nil, args)
```

err is ignored here.

---

Ah, that call shouldn't be necessary anymore.